### PR TITLE
Array plots

### PR DESCRIPTION
**Pull request by jansilar**
Arrays are collapsable in the tree view
New Plot array and Plot array parametric buttons and plots
Time slider controlling time of array (parametric) plots

### CODE REVIEW

**Comment by adeas31** [comment]
@jansilar Can you also squash these commits?

**Comment by jansilar** [comment]
Yes, the history is ugly. The problem is that it contains several merges from master so the squash could mess thinks up. Probably the 6 my consequent commits may be squashed safely. Is it right, even though merges from master will follow this squashed commit?

**Comment by sjoelund** [comment]
It's pretty much always safe to rebase squash all commits into one. Because all it does it take the same state of all files that would happen if you merged it, and make a single commit on top of the current master instead of a series of merges.

**Comment by jansilar** [comment]
OK, its squashed. I didn't know git can resolve merges during rebase.

**Comment by adeas31** [comment]
@jansilar There is small bug. When you switch between normal plot and array plot window the checked variables are cleared. Here are steps to reproduce it,

1. Simulate a normal Modelica model and plot a variable lets say variable `x`.
2. Simulate a pde model and plot an array variable lets say variable `y`.
3. Until now you have two plot windows one is normal plot window and the other is array plot window. Array plot window is the current one and you also see variable `y` checked.
4. Switch to normal plot window by pressing `ctrl+tab`.
5. You will see variable `x` checked.
6. Now switch back to array plot window. The variable `u` is shown plotted but is not checked in the variables browser.

**Comment by jansilar** [comment]
@adeas31 Thanks for testing. It's fixed now.

**Comment by adeas31** [comment]
There are still some issues. Here are steps to reproduce it,

1. Simulate a pde model and plot an array variable lets say variable `x`. OMEdit will plot the variable on an array plot window.
2. Plot the array index 1 i.e., `x[1]`. OMEdit will open a new plot window and plot `x[1]`.
3. Now at this time you will see both `x` and `x[1]` checked. Only `x[1]` should be checked.

**Comment by jansilar** [comment]
@adeas31 I can't figure it out, could you help? So far I found out this:
When new plot(array) window is opened in step 2. and 3. the signal subWindowActivated(...) should be raised and slot updateVariablesTree(...) should be called. But it is not called - breakpoint in it is not hit. The signal is inherited from QMdiArea so I don't know what to do about that.

File: OMEdit/OMEditGUI/Plotting/VariablesWidget.cpp [hunk 1]
```diff
@@ -1237,13 +1282,33 @@ void VariablesWidget::plotVariables(const QModelIndex &index, qreal curveThickne
...
+    // then create a new plot window.
+    if (!pVariablesTreeItem->isMainArray() &&
+            (!pPlotWindow || pPlotWindow->getPlotType()==PlotWindow::PLOTARRAY
+             || pPlotWindow->getPlotType()== PlotWindow::PLOTARRAYPARAMETRIC))
+    {
       bool state = MainWindow::instance()->getPlotWindowContainer()->blockSignals(true);
       MainWindow::instance()->getPlotWindowContainer()->addPlotWindow();
       MainWindow::instance()->getPlotWindowContainer()->blockSignals(state);
       pPlotWindow = MainWindow::instance()->getPlotWindowContainer()->getCurrentWindow();
     }
```
Comment (adeas31) [inline OMEdit/OMEditGUI/Plotting/VariablesWidget.cpp]: In the above block we block the signals that is why `subWindowActivated` signal is not raised. You can remove the block with the following code,
```
MainWindow::instance()->getPlotWindowContainer()->addPlotWindow();
bool state = MainWindow::instance()->getPlotWindowContainer()->blockSignals(true);
mpVariablesTreeModel->setData(index, Qt::Checked, Qt::CheckStateRole);
MainWindow::instance()->getPlotWindowContainer()->blockSignals(state);
pPlotWindow = MainWindow::instance()->getPlotWindowContainer()->getCurrentWindow();
```
The same goes for the plot parametric block below.

**Comment by jansilar** [comment]
Fixed. Thanks.

**Pull request merged**